### ISSUE TITLE
SamlResponse issuer verification

### ISSUE THREAD

**Issue by mauromol** [onelogin/java-saml#325]
Currently `SamlResponse` during its validity check ensures that the issuer of the response and the issuer of the assertion (with the constraint that just one assertion is present) have both a value equal to the IdP entity id.
This is fine, however no check is made against the format of such issuers. Specification says that the issuer format must be `urn:oasis:names:tc:SAML:2.0:nameid-format:entity` and that, in this case, no qualifier must be specified. But what about a malicious/erroneous response?
Perhaps java-saml could add a check that ensures that the format is either omitted or `urn:oasis:names:tc:SAML:2.0:nameid-format:entity`? And perhaps that no qualifier (and maybe also no SP qualifier) is specified as well?

The Italian SPID specification requires the former check (the one about the format) is made, although we could say that this might be an excess of rigor. 

What do you think?

**Comment by pitbulk** [comment]
I don't think this one gonna be required. At the end the toolkit make nothing with the  "urn:oasis:names:tc:SAML:2.0:nameid-format:entity" format value

**Comment by mauromol** [comment]
True, there probably is no impact from a functional point of view. I'm just wondering about security implications... Could this missing check bring an issue? I really don't know. After all, why does SAML 2.0 specification require this?

**Comment by pitbulk** [comment]
I dont think so, the format on the NameID, Attributes from the AttributeStatement, ... are used by the SP in order to know how to handle the value of such attribute, but in that specific case, we always assume the value is a string to be compared with the EntityID registered.

**Comment by mauromol** [comment]
What about an opt-in option "check response issuers format" configurable in settings, false by default?

**Comment by pitbulk** [comment]
I don't see justified on this case the addition of a new setting

**Comment by mauromol** [comment]
Ok, so do you prefer to just ignore this strict SAML compliance checking and let consumers implement it if desired?

With #318 and #352 that would be feasible.

**Comment by pitbulk** [comment]
Yes
